presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -180,10 +187,19 @@ public GlueHiveMetastore(
     private static AWSGlueAsync createAsyncGlueClient(GlueHiveMetastoreConfig config)
     {
         ClientConfiguration clientConfig = new ClientConfiguration().withMaxConnections(config.getMaxGlueConnections());
+        if (config.getGlueEndpointUrl().isPresent()) {
+            checkArgument(config.getGlueRegion().isPresent(), "Glue region must be set when Glue endpoint URL is set");
+            // Certificate host name validation fails if we are connecting with VPC endpoint host name
```

Comment:
I don't understand this. The AWS documentation doesn't mention anything about this: https://docs.aws.amazon.com/glue/latest/dg/vpc-endpoint.html

The SDK Javadoc for `com.amazonaws.sdk.disableCertChecking` says

> Disable validation of server certificates when using the HTTPS protocol. This should ONLY be used to do quick smoke tests against endpoints which don't yet have valid certificates; it should NEVER be used in production.

As a general security practice, we don't add code to Presto to allow users to defeat security mechanisms.

If this actually is needed, please provide specific AWS documentation about why. And in that case, it needs to be a separate configuration property, not tied to setting the endpoint URL.

Reply:
> I don't understand this. The AWS documentation doesn't mention anything about this: https://docs.aws.amazon.com/glue/latest/dg/vpc-endpoint.html
> 
> The SDK Javadoc for `com.amazonaws.sdk.disableCertChecking` says
> 
> > Disable validation of server certificates when using the HTTPS protocol. This should ONLY be used to do quick smoke tests against endpoints which don't yet have valid certificates; it should NEVER be used in production.

We are not setting com.amazonaws.sdk.disableCertChecking. We disable host name validation only for Glue client and only if we're using endpoint.

> If this actually is needed, please provide specific AWS documentation about why. And in that case, it needs to be a separate configuration property, not tied to setting the endpoint URL.

I want it to be tied to endpoint and not as a general setting (for the reason you stated). 
The Glue service provides certificate for it's public name (e.g. glue.us-east-2.amazonaws.com). VPC endpoints have its own DNS names, specific for user/vpc. Connecting to such endpoint will fail with standard host name verifier.

Other possible options:
1. We could not provide this functionality at all and require enabling Private DNS on AWS vpc endpoint. But this requires some specific settings on VPC that we may not want to impose on users.
2. We could enable it only if required because user cannot enable Private DNS (adding configuration property). 
3. We could create more strict hostname verifier, which actually verifies that we connect to region-specific glue service.
4. We can combine 2 and 3...

Reply:
BTW - the same behaviour is using e.g. aws-cli:

```
# aws --endpoint-url https://vpce-03284955d8f8f36ae-hug1rabn.glue.us-east-2.vpce.amazonaws.com/   glue get-databases --region us-east-2

SSL validation failed for https://vpce-03284955d8f8f36ae-hug1rabn.glue.us-east-2.vpce.amazonaws.com/ ("hostname 'vpce-03284955d8f8f36ae-hug1rabn.glue.us-east-2.vpce.amazonaws.com' doesn't match 'glue.us-east-2.amazonaws.com'",)
```

Reply:
Removed SSL disable code, we will use proxy instead.